src/output.cpp

```diff
@@ -1112,8 +1112,18 @@ std::string JsonOutput::value_to_str(BPFtrace &bpftrace,
                                      uint32_t div,
                                      bool is_map_key) const
 {
-  auto str = Output::value_to_str(
-      bpftrace, type, value, is_per_cpu, div, is_map_key);
+  std::string str;
+
+  switch (type.GetTy()) {
```

Comment:
this is super nitpicky so I won't block but I think perhaps moving the pointer and reference printing out of the base Output class `value_to_str` and putting them (like you have it here) in `TextOutput::value_to_str` so it's clear that there is a difference in the output type.